packages/core-components/src/components/DismissableBanner/DismissableBanner.tsx

```diff
@@ -73,8 +73,10 @@ const useStyles = makeStyles(
         color: theme.palette.banner.link,
       },
     },
-    button: {
-      color: theme.palette.banner.closeButtonColor ?? 'inherit',
+    button_info: { color: theme.palette.banner.closeButtonColor ?? 'inherit' },
+    button_error: { color: theme.palette.banner.closeButtonColor ?? 'inherit' },
+    button_warning: {
+      color: theme.palette.banner.closeButtonColorDark ?? 'inherit',
     },
```

Comment:
This encodes particular light/dark combinations for different variants, which then ends up not being themeable. I think it's worth spending a bit of time here thinking about if there's some way we can achieve this without needing to encode too much theming logic within the component itself.